Fix egress CIDR policy enforcement

tgraf opened the pull request.

tgraf committed:
bpf: Fix egress CIDR policy enforcement

The existing egress CIDR enforcement lookup was performed correctly but
relied on a later policy check enforcing a drop and thus only marked the
packet to skip the policy check instead of dropping it directly. The
latter policy check was removed and since broke the egress CIDR policy
enforcement.

The CI test is in-effective and thus did not catch this regression. This
commit fixes the bug, the CI test will be fixed in a separate commit.

Fixes: #3345
Fixes: #3340

Signed-off-by: Thomas Graf <thomas@cilium.io>

rlenglet commented on bpf/bpf_overlay.c:
```diff
@@ -1,4 +1,4 @@
-/*
+ /*
```
Remove leading space.

tgraf replied:
fixed

tgraf committed:
bpf: Fix return value of lpm{46}_ingress_lookup

When policy enforcement is disabled, these lookup calls should always
return a positive value to indicate a match. This bug had no effect
so far as calls to these functions are currently protected by defines
only defined when policy enforcement is enabled.

Signed-off-by: Thomas Graf <thomas@cilium.io>
